src/main/java/wedlog/address/model/tag/GuestDietaryPredicate.java

```diff
@@ -0,0 +1,50 @@
+package wedlog.address.model.tag;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import wedlog.address.commons.util.ToStringBuilder;
+import wedlog.address.model.person.Guest;
+
+/**
+ * Tests that a {@code Person}'s {@code Dietary Requirement} matches any of the keywords given.
+ */
+public class GuestDietaryPredicate implements Predicate<Guest> {
+    private final List<String> keywords;
+
+    /**
+     * Constructor for GuestDietaryPredicate.
+     */
+    public GuestDietaryPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Guest guest) {
+        return !keywords.isEmpty() && keywords.get(0).isEmpty()
+                ? guest.getDietaryRequirements().isEmpty()
+                : keywords.stream().anyMatch(keyword -> guest.getDietaryRequirements().stream()
+                .anyMatch(dr -> DietaryRequirement.isValidDietaryRequirement(keyword)
+                        && dr.equals(new DietaryRequirement(keyword))));
+    }
```

Comment:
actually on this note, do we want to return an empty guest list if the input is invalid, or do we want to throw a parse exception when it does not have the correct format?

this could potentially lead to a change in our UG for appendix A for the allowed values for each parameter

Reply:
Invalid input should still trigger an error message per the current implementation